### PR TITLE
firmware: Align firmware blobs.

### PR DESCRIPTION
**Pull request by iabdalkader**
* The firmware blobs need to aligned for loading with the DMA at least on the stm32 port, since the big_const symbol was removed this is no longer controllable by the linker script.
* Fixes firmware loading on Portenta, Giga, Nicla.

### CODE REVIEW

**Comment by peterharperuk** [comment]
Seems reasonable. I guess we broke that when moving to header files.

**Comment by dpgeorge** [comment]
It is possible to do this using the linker script, you can explicitly align things using `. = ALIGN(4);` and then place the firmware blobs right after that align.

That said, I think it's cleaner to do this alignment as it's done in this PR, in the C code.

But it should be configurable.

File: firmware/w43439A0_7_95_49_00_combined.h [hunk 1]
```diff
@@ -1,4 +1,4 @@
-static const unsigned char w43439A0_7_95_49_00_combined[] = {
+static const unsigned char w43439A0_7_95_49_00_combined[] __attribute__((aligned(4))) = {
```
Comment (dpgeorge) [inline firmware/w43439A0_7_95_49_00_combined.h:1]: I suggest making this `CYW43_FIRMWARE_ATTRIBUTE` and then a port can do:
```c
#define CYW43_FIRMWARE_ATTRIBUTE __attribute__((aligned(4)))
```

**Comment by iabdalkader** [comment]
> It is possible to do this using the linker script, you can explicitly align things using `. = ALIGN(4);` and then place the firmware blobs right after that align.

Yes but you will need to know the name of the firmware blob and to change back all the linker script.

> That said, I think it's cleaner to do this alignment as it's done in this PR, in the C code.

I had some doubts about one alignment fit all, but I checked and this works for F7 and H7, that said I will still do the suggested change above.

**Comment by dpgeorge** [comment]
I renamed the config option to `CYW43_RESOURCE_ATTRIBUTE` and applied it also to the NVRAM (which was already aligned, so this makes it consistent with the other blobs).

Thanks for finding this issue!

**Pull request merged**